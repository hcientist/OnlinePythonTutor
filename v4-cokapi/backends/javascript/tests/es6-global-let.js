var a = 5;
var b = 10;

if (a === 5) {
  let a = 4; // The scope is inside the if-block
  var b = 1; // The scope is inside the function

  console.log(a);  // 4
  console.log(b);  // 1
} 

console.log(a); // 5
console.log(b); // 1
