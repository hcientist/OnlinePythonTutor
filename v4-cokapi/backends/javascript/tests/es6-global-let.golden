{ code: 'var a = 5;\nvar b = 10;\n\nif (a === 5) {\n  let a = 4; // The scope is inside the if-block\n  var b = 1; // The scope is inside the function\n\n  console.log(a);  // 4\n  console.log(b);  // 1\n} \n\nconsole.log(a); // 5\nconsole.log(b); // 1',
  trace: 
   [ { stdout: '',
       func_name: '',
       stack_to_render: [],
       globals: 
        { a: [ 'JS_SPECIAL_VAL', 'undefined' ],
          b: [ 'JS_SPECIAL_VAL', 'undefined' ] },
       ordered_globals: [ 'a', 'b' ],
       line: 1,
       col: 0,
       event: 'step_line',
       heap: {} },
     { stdout: '',
       func_name: '',
       stack_to_render: [],
       globals: { a: 5, b: [ 'JS_SPECIAL_VAL', 'undefined' ] },
       ordered_globals: [ 'a', 'b' ],
       line: 2,
       col: 0,
       event: 'step_line',
       heap: {} },
     { stdout: '',
       func_name: '',
       stack_to_render: [],
       globals: { a: 5, b: 10 },
       ordered_globals: [ 'a', 'b' ],
       line: 4,
       col: 0,
       event: 'step_line',
       heap: {} },
     { stdout: '',
       func_name: '',
       stack_to_render: [],
       globals: { a: 5, b: 10, 'a (block 1)': [ 'JS_SPECIAL_VAL', 'undefined' ] },
       ordered_globals: [ 'a', 'b', 'a (block 1)' ],
       line: 5,
       col: 2,
       event: 'step_line',
       heap: {} },
     { stdout: '',
       func_name: '',
       stack_to_render: [],
       globals: { a: 5, b: 10, 'a (block 1)': 4 },
       ordered_globals: [ 'a', 'b', 'a (block 1)' ],
       line: 6,
       col: 2,
       event: 'step_line',
       heap: {} },
     { stdout: '',
       func_name: '',
       stack_to_render: [],
       globals: { a: 5, b: 1, 'a (block 1)': 4 },
       ordered_globals: [ 'a', 'b', 'a (block 1)' ],
       line: 8,
       col: 2,
       event: 'step_line',
       heap: {} },
     { stdout: '4\n',
       func_name: '',
       stack_to_render: [],
       globals: { a: 5, b: 1, 'a (block 1)': 4 },
       ordered_globals: [ 'a', 'b', 'a (block 1)' ],
       line: 9,
       col: 2,
       event: 'step_line',
       heap: {} },
     { stdout: '4\n1\n',
       func_name: '',
       stack_to_render: [],
       globals: { a: 5, b: 1 },
       ordered_globals: [ 'a', 'b' ],
       line: 12,
       col: 0,
       event: 'step_line',
       heap: {} },
     { stdout: '4\n1\n5\n',
       func_name: '',
       stack_to_render: [],
       globals: { a: 5, b: 1 },
       ordered_globals: [ 'a', 'b' ],
       line: 13,
       col: 0,
       event: 'step_line',
       heap: {} },
     { stdout: '4\n1\n5\n1\n',
       func_name: '',
       stack_to_render: [],
       globals: { a: 5, b: 1 },
       ordered_globals: [ 'a', 'b' ],
       line: 13,
       col: 20,
       event: 'step_line',
       heap: {} } ] }
